db/schema_applier.hh

```diff
@@ -31,4 +31,66 @@ future<> merge_schema(sharded<db::system_keyspace>& sys_ks, distributed<service:
 // of feature_service and schema tables.
 future<> recalculate_schema_version(sharded<db::system_keyspace>& sys_ks, distributed<service::storage_proxy>& proxy, gms::feature_service& feat);
 
+enum class table_kind { table, view };
+
+struct table_selector {
+    bool all_in_keyspace = false; // If true, selects all existing tables in a keyspace plus what's in "tables";
+    std::unordered_map<table_kind, std::unordered_set<sstring>> tables;
+
+    table_selector& operator+=(table_selector&& o);
+    void add(table_kind t, sstring name);
+    void add(sstring name);
+};
+
+struct schema_complete_view {
+    schema_tables::schema_result keyspaces;
+    std::map<table_id, schema_mutations> tables;
+    schema_tables::schema_result types;
+    std::map<table_id, schema_mutations> views;
+    schema_tables::schema_result functions;
+    schema_tables::schema_result aggregates;
+    schema_tables::schema_result scylla_aggregates;
+};
+
+class schema_applier {
+    using keyspace_name = sstring;
+
+    sharded<service::storage_proxy>& _proxy;
+    sharded<db::system_keyspace>& _sys_ks;
+    bool _reload;
+
+    std::set<sstring> _keyspaces;
+    std::unordered_map<keyspace_name, table_selector> _affected_tables;
+    bool _has_tablet_mutations = false;
```

Comment:
Since there's a ctor, I'd move initializing this member to the ctor.

Reply:
cpp core guidlines seem to disagree: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c48-prefer-default-member-initializers-to-member-initializers-in-constructors-for-constant-initializers

Reply:
CPP core guidelines present their argument on examples using 2 ctors, whereas here we only have 1 (of course this may change in the future). Mine & @KrzaQ's reasoning was that in case there's only 1 ctor it's more readable to have everything initialized in the ctor, rather than to init some members in the ctor, and others with their definition.

Reply:
guidelines are generic, not constrained to situations given in example, so as I understand applies to 1,2,3 and any number of constructors.